src/html.c

```diff
@@ -63,10 +63,14 @@ static bool S_put_footnote_backref(cmark_html_renderer *renderer, cmark_strbuf *
   if (renderer->written_footnote_ix >= renderer->footnote_ix)
     return false;
   renderer->written_footnote_ix = renderer->footnote_ix;
+  char m[32];
+  snprintf(m, sizeof(m), "%d", renderer->written_footnote_ix);
```

Comment:
Just as a note, my concern would be about snprintf implementations that don't nul terminate properly. If it were possible to get a 32byte width integer value out of the backref indexes (highly unlikely imo), we are not guaranteed a nul terminated string in m, which may lead to memory info leaks as cmark_strbuf_puts is strlen based (i.e. it will think len is 32 + whatever offset the first nul byte is located in adjacent memory). AFAICT everything that we build against in our test matrix is C99 conformant (but e.g. old visual studio versions pre-2015 were not iirc). /cc @kevinbackhouse for thoughts.

Reply:
If we wanted to be abundantly careful re: C99 compliance we could maybe do something like:

```suggestion
  char m[32];
  memset(m, 0, sizeof(m));
  snprintf(m, sizeof(m)-1, "%d", renderer->written_footnote_ix);
```

Reply:
As per @kevinbackhouse %d is not able to print an integer of that width, so this should be fine.

Reply:
Thanks for raising this! To clarify, by “this should be fine”, did you mean that the code in this PR is fine as-is, or that the snippet you provided above sufficiently addresses the issue you noted?

I’m happy to go with whatever you recommend. Should we change this elsewhere? I was following the examples below, and I don’t know C or this codebase well enough to understand if these are categorically different in some way:

https://github.com/github/cmark-gfm/blob/eb32891cd817aff1772d6d2783c011e29422cc9d/src/html.c#L74-L75

and 

https://github.com/github/cmark-gfm/blob/eb32891cd817aff1772d6d2783c011e29422cc9d/src/html.c#L432-L433